src/Components/test/E2ETest/Tests/ThreadingAppTest.cs

```diff
@@ -0,0 +1,130 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests;
+
+public class ThreadingAppTest
+    : ServerTestBase<BlazorWasmTestAppFixture<ThreadingApp.Program>>, IDisposable
+{
+    public ThreadingAppTest(
+        BrowserFixture browserFixture,
+        BlazorWasmTestAppFixture<ThreadingApp.Program> serverFixture,
+        ITestOutputHelper output)
+        : base(browserFixture, serverFixture, output)
+    {
+    }
+
+    protected override void InitializeAsyncCore()
+    {
+        Navigate("/", noReload: true);
+        WaitUntilLoaded();
+    }
+
+    [Fact]
+    public void HasTitle()
+    {
+        Assert.Equal("Blazor standalone", Browser.Title);
+    }
+
+    [Fact]
+    public void HasHeading()
+    {
+        Assert.Equal("Hello, world!", Browser.Exists(By.TagName("h1")).Text);
+    }
+
+    [Fact]
+    public void NavMenuHighlightsCurrentLocation()
+    {
+        var activeNavLinksSelector = By.CssSelector(".sidebar a.active");
+        var mainHeaderSelector = By.TagName("h1");
+
+        // Verify we start at home, with the home link highlighted
+        Assert.Equal("Hello, world!", Browser.Exists(mainHeaderSelector).Text);
+        Assert.Collection(Browser.FindElements(activeNavLinksSelector),
+            item => Assert.Equal("Home", item.Text.Trim()));
+
+        // Click on the "counter" link
+        Browser.Exists(By.LinkText("Counter")).Click();
+
+        // Verify we're now on the counter page, with that nav link (only) highlighted
+        Assert.Equal("Counter", Browser.Exists(mainHeaderSelector).Text);
+        Assert.Collection(Browser.FindElements(activeNavLinksSelector),
+            item => Assert.Equal("Counter", item.Text.Trim()));
+
+        // Verify we can navigate back to home too
+        Browser.Exists(By.LinkText("Home")).Click();
+        Assert.Equal("Hello, world!", Browser.Exists(mainHeaderSelector).Text);
+        Assert.Collection(Browser.FindElements(activeNavLinksSelector),
+            item => Assert.Equal("Home", item.Text.Trim()));
+    }
+
```

Comment:
Could we have a couple of additional tests for Blazor JS interop as well as the runtime JS interop? We should probably have a flavor an operation that completes synchronously (like a function that wraps the result in a task) as well as an operation that completes asynchronously for each of the operations.

Reply:
not on this PR, sorry